python/src/aiconfig/editor/client/src/components/prompt/prompt_input/PromptInputJSONRenderer.tsx

```diff
@@ -0,0 +1,76 @@
+import { JSONValue, PromptInput } from "aiconfig";
+import { memo, useCallback } from "react";
+import JSONRenderer from "../../JSONRenderer";
+
+type Props = {
+  input: PromptInput;
+  onChangeInput: (value: PromptInput) => void;
+};
+
+const PROMPT_INPUT_SCHEMA = {
+  anyOf: [
+    {
+      type: "object",
+      additionalProperties: {},
+      properties: {
+        data: {
+          description:
+            "Input to the model. This can represent a single input, or multiple inputs.\nThe structure of the data object is up to the ModelParser. Attachments field should be leveraged for non-text inputs (e.g. image, audio).",
+        },
+        attachments: {
+          description:
+            "Used to include non-text inputs (e.g. image, audio) of specified MIME types in the prompt",
+          type: "array",
+          items: {
+            $ref: "#/definitions/Attachment",
+          },
+        },
+      },
+    },
+    {
+      type: "string",
+    },
+  ],
+  definitions: {
+    Attachment: {
+      description: "Data of specified MIME type to attach to a prompt",
+      type: "object",
+      required: ["data"],
+      properties: {
+        mime_type: {
+          description:
+            "MIME type of the attachment. If not specified, the MIME type will be assumed to be text/plain.",
+          type: "string",
+        },
+        data: {
+          description: "Data representing the attachment",
+        },
+        metadata: {
+          description: "Attachment metadata.",
+          type: "object",
+          additionalProperties: {},
+        },
+      },
+    },
+  },
+};
```

Comment:
Is there a way to sub-reference our own schema so that this is never out of sync? Ex: `$ref/definitions/InputPrompt`

Reply:
There *should* be, but I couldn't get it to work within a reasonable timebox. See comment in JSONEditor:

>         // schema: {
        //     $ref: "#/definitions/PromptInput",
        // }
        // TODO: Figure out how to reference PromptInput definition from the uri schema
        // Getting the following error:
        // $ref '/definitions/PromptInput' in 'https://json.schemastore.org/aiconfig-1.0' can not be resolved.(768)